activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java

```diff
@@ -630,10 +650,322 @@ public void testTwoSubProcesses() {
+                                     tuple(SEQUENCE_FLOW_TAKEN,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY),
+                                     tuple(ACTIVITY_STARTED,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY),
+                                     tuple(TASK_CANDIDATE_GROUP_ADDED,null, null, null),
+                                     tuple(TASK_CREATED,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY)
+                    );
+            
+            
+            assertThat(receivedEvents)
+            .filteredOn(event -> ACTIVITY_STARTED.equals(event.getEventType()))
+            .extracting(event -> ((CloudBPMNActivityStartedEvent) event).getEntity().getActivityType(),
+                        event -> ((CloudBPMNActivityStartedEvent) event).getEntity().getActivityName()
+                        )
+            .containsExactly(tuple("startEvent", null),
+                             tuple("subProcess", "subProcess"),
+                             tuple("startEvent", null),
+                             tuple("userTask", "Task in subprocess")
+                             );
+            
+
+        });
+       
+        ResponseEntity<PagedResources<CloudTask>> tasks = processInstanceRestTemplate.getTasks(processInstance);
+        Task task = tasks.getBody().iterator().next();
+        
+        String subProcessInstanceId = task.getProcessInstanceId();
+        assertThat(subProcessInstanceId).isNotNull();
+        assertThat(subProcessInstanceId).isEqualTo(processInstanceId);
+        
+        //when
+        taskRestTemplate.claim(task);
+        
+        await().untilAsserted(() -> {
+            
+            assertThat(streamHandler.getReceivedHeaders()).containsKeys(ALL_REQUIRED_HEADERS);
+            List<CloudRuntimeEvent<?, ?>> receivedEvents = streamHandler.getLatestReceivedEvents();
+            
+            assertThat(receivedEvents)
+                    .extracting(CloudRuntimeEvent::getEventType,
+                                CloudRuntimeEvent::getProcessInstanceId,
+                                CloudRuntimeEvent::getParentProcessInstanceId,
+                                CloudRuntimeEvent::getProcessDefinitionKey
+                                )
+                    .containsExactly(tuple(TASK_ASSIGNED,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY),
+                                     tuple(TASK_UPDATED,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY));
+        });
+
+        
+        //when
+        taskRestTemplate.complete(task);
+        
+        //Check we have two processes and one of them is callActivity process
+        // when
+        List<org.activiti.engine.runtime.ProcessInstance> processes = runtimeService.createProcessInstanceQuery()
```

Comment:
Prefer to use the API than internal `RuntimeService`: you can use `ProcessInstanceRestTemplate.getSubprocesses(processInstanceId)` for this.